imap_processing/codice/codice_l1a_hi_counters_aggregated.py

```diff
@@ -0,0 +1,181 @@
+
+    # The decompressed data in the shape of (epoch, n). Then reshape later.
+    decompressed_data = [
+        decompress(
+            packet_data[:byte_count],
+            compression_algorithm,
+        )
+        for (packet_data, byte_count) in zip(
+            binary_data_list, byte_count_list, strict=False
+        )
+    ]
+    counters_data = np.array(decompressed_data, dtype=np.uint32).reshape(
+        -1, num_variables
+    )
+
+    # ========= Get Epoch Time Data ===========
+    # Epoch center time and delta
+    epoch_center, deltas = get_codice_epoch_time(
+        unpacked_dataset["acq_start_seconds"].values,
+        unpacked_dataset["acq_start_subseconds"].values,
+        unpacked_dataset["spin_period"].values,
+        view_tab_obj,
+    )
+
+    # ========== Initialize CDF Dataset with Coordinates ===========
+    cdf_attrs = ImapCdfAttributes()
+    cdf_attrs.add_instrument_global_attrs("codice")
+    cdf_attrs.add_instrument_variable_attrs("codice", "l1a")
+
+    l1a_dataset = xr.Dataset(
+        coords={
+            "epoch": xr.DataArray(
+                met_to_ttj2000ns(epoch_center),
+                dims=("epoch",),
+                attrs=cdf_attrs.get_variable_attributes("epoch", check_schema=False),
+            ),
+            "epoch_delta_minus": xr.DataArray(
+                deltas,
+                dims=("epoch",),
+                attrs=cdf_attrs.get_variable_attributes(
+                    "epoch_delta_minus", check_schema=False
+                ),
+            ),
+            "epoch_delta_plus": xr.DataArray(
+                deltas,
+                dims=("epoch",),
+                attrs=cdf_attrs.get_variable_attributes(
+                    "epoch_delta_plus", check_schema=False
+                ),
+            ),
+        },
+        attrs=cdf_attrs.get_global_attributes(logical_source_id),
+    )
+
+    # Add first few unique variables
+    l1a_dataset["spin_period"] = xr.DataArray(
+        unpacked_dataset["spin_period"].values * constants.SPIN_PERIOD_CONVERSION,
+        dims=("epoch",),
+        attrs=cdf_attrs.get_variable_attributes("spin_period"),
+    )
+    l1a_dataset["data_quality"] = xr.DataArray(
+        unpacked_dataset["suspect"].values,
+        dims=("epoch",),
+        attrs=cdf_attrs.get_variable_attributes("data_quality"),
+    )
+
+    # Finally, add species data variables and their uncertainties
+    for idx, species in enumerate(non_reserved_variables):
+        # Get CDF attrs
+        if species.startswith("reserved"):
+            # extract reserved index
+            reserved_index = species.replace("reserved", "")
+            attrs = cdf_attrs.get_variable_attributes("reserved")
+            # Apply index replacement
+            attrs = apply_replacements_to_attrs(attrs, {"{index}": reserved_index})
+        else:
+            attrs = cdf_attrs.get_variable_attributes(species)
+        l1a_dataset[species] = xr.DataArray(
```

Comment:
Could you add an extra space here, otherwise this blended together to my eyes and I thought it was a part of the else-block with the wrapping applied below.
```suggestion

        l1a_dataset[species] = xr.DataArray(
```